File: DFS.cpp
#include <iostream>
#include <vector>
using namespace std;

void dfs(int node, vector<vector<int>> &adj, vector<int> &vis, vector<int> &res) {
    vis[node] = 1;
    res.push_back(node);

    for (auto it : adj[node]) {
        if (!vis[it]) {
            dfs(it, adj, vis, res);
        }
    }
}

vector<int> dfsOfGraph(int V, vector<vector<int>> &adj, int start) {
    vector<int> vis(V + 1, 0);  
    vector<int> res;

    dfs(start, adj, vis, res);
    return res;
}

int main() {
    int V, E;
    cout << "Enter number of vertices: ";
    cin >> V;

    cout << "Enter number of edges: ";
    cin >> E;

    vector<vector<int>> adj(V + 1); 

    cout << "Enter edges (u v):\n";
    for (int i = 0; i < E; i++) {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);   
    }

    int start;
    cout << "Enter starting vertex: ";
    cin >> start;

    vector<int> dfsTraversal = dfsOfGraph(V, adj, start);

    cout << "DFS Traversal: ";
    for (int node : dfsTraversal) {
        cout << node << " ";
    }

    return 0;
}

File: DFS.py
def dfs(node, adj, visited, res):
    visited[node] = True
    res.append(node)

    for it in adj[node]:
        if not visited[it]:
            dfs(it, adj, visited, res)


def dfs_of_graph(V, adj, start):
    visited = [False] * (V + 1)
    res = []

    dfs(start, adj, visited, res)
    return res


V = int(input("Enter number of vertices: "))
E = int(input("Enter number of edges: "))

adj = [[] for _ in range(V + 1)]

print("Enter edges (u v):")
for _ in range(E):
    u, v = map(int, input().split())
    adj[u].append(v)
    adj[v].append(u)

start = int(input("Enter starting vertex: "))

dfs_traversal = dfs_of_graph(V, adj, start)

print("DFS Traversal:", *dfs_traversal)
